### PR TITLE
NEWS.md: add release notes for v0.18.0

### PR DESCRIPTION
**Pull request by garlick**
First attempt at v0.18.0 release notes.  Others feel free to push to this branch!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,51 @@
+flux-core version 0.18.0 - 2020-07-29
+-------------------------------------
+
+This release features a more compact default representation for Flux job ID's,
```
Comment (grondo) [inline NEWS.md:4]: s/ID's/IDs/ (or just JOBIDs)

Comment (garlick) [reply NEWS.md:4]: Argh I always do that!  Thanks.

**Comment by dongahn** [comment]
LGTM too. I really like the feature highlights described in the beginning @garlick!

**Comment by garlick** [comment]
> I really like the feature highlights described in the beginning @garlick!

Me too!  Credit goes to @grondo for that idea.

**Pull request merged**